Assets/Scripts/PlayerCameraScript.cs

```diff
@@ -6,15 +6,35 @@ public class PlayerCamera : MonoBehaviour
 {
 
     public Transform playerTransform;
+    public GameController gameController;
+    public bool isMoving;
+    private bool isAnimating;
+    public Vector3 target;
+    Vector3 moveVelocity;
 
-    // Update is called once per frame
     void LateUpdate()
     {
-        this.transform.position = new Vector3(playerTransform.position.x, playerTransform.position.y, this.transform.position.z);
+        if (!isAnimating & !isMoving){
```

Comment:
You should use '&&' here instead of '&'. The single & is bitwise, I'm not exactly sure what it does but I think it has the potentially to behave in a weird way if you use it like this.